api/src/main/java/org/openmrs/module/dhisconnector/api/impl/DHISConnectorServiceImpl.java

```diff
@@ -838,102 +835,50 @@ private boolean mappingsHasGUID(List<DHISMapping> mappings, String GUID) {
 	}
 	
-	@SuppressWarnings("rawtypes")
 	@Override
-	public String uploadMappings(MultipartFile mapping) {
-		String msg = "";
-		String tempFolderName = OpenmrsUtil.getApplicationDataDirectory() + DHISCONNECTOR_TEMP_FOLDER + File.separator;
-		String mappingFolderName = OpenmrsUtil.getApplicationDataDirectory() + DHISCONNECTOR_MAPPINGS_FOLDER
-		        + File.separator;
-		String mappingName = mapping.getOriginalFilename();
-
-		File mappingsFolder = new File(mappingFolderName);
-		mappingsFolder.mkdirs();
-
-		if (!mappingsFolder.isDirectory()) {
-			return Context.getMessageSourceService().getMessage("dhisconnector.uploadMapping.directoryNotCreated");
-		}
+	public String importMappings (MultipartFile mappingBundle) throws IOException {
+		String sourceDirectory = OpenmrsUtil.getApplicationDataDirectory() + DHISCONNECTOR_MAPPINGS_FOLDER + File.separator;
+		String tempDirectory = OpenmrsUtil.getApplicationDataDirectory() + DHISCONNECTOR_TEMP_FOLDER + File.separator;
+		(new File(sourceDirectory)).mkdirs();
+		(new File(tempDirectory)).mkdirs();
+		ObjectMapper mapper = new ObjectMapper();
+		XmlMapper xmlMapper = new XmlMapper();
 
-		if (mappingName.endsWith(".zip")) {
-			boolean allFailed = true;
-			File tempMappings = new File(tempFolderName + mappingName);
-			
-			(new File(tempFolderName)).mkdirs();
-			try {
-				mapping.transferTo(tempMappings);
-				
-				try {
-					ZipFile zipfile = new ZipFile(tempMappings);
-					
-					for (Enumeration e = zipfile.entries(); e.hasMoreElements();) {
-						ZipEntry entry = (ZipEntry) e.nextElement();
-						
-						if (entry.isDirectory()) {
-							System.out.println(
-							    "Incorrect file (Can't be a folder instead): " + entry.getName() + " has been ignored");
-						} else if (entry.getName().endsWith(DHISCONNECTOR_MAPPING_FILE_SUFFIX)) {
-							File outputFile = new File(mappingFolderName, entry.getName());
-							
-							if (outputFile.exists()) {
-								System.out.println("File: " + outputFile.getName() + " already exists and has been ignored");
-							} else {
-								BufferedInputStream inputStream = new BufferedInputStream(zipfile.getInputStream(entry));
-								BufferedOutputStream outputStream = new BufferedOutputStream(
-								        new FileOutputStream(outputFile));
-								
-								try {
-									System.out.println("Extracting: " + entry);
-									IOUtils.copy(inputStream, outputStream);
-									allFailed = false;
-								}
-								finally {
-									outputStream.close();
-									inputStream.close();
-								}
-							}
-						} else {
-							System.out.println("Incorrect file: " + entry.getName() + " has been ignored");
-						}
-					}
-					if (!allFailed) {
-						msg = Context.getMessageSourceService().getMessage("dhisconnector.uploadMapping.groupSuccess");
-					} else {
-						msg = Context.getMessageSourceService().getMessage("dhisconnector.uploadMapping.allFailed");
+		if (mappingBundle.getOriginalFilename().endsWith("zip")) {
```

Comment:
thanks @Gravewalker666 ,don't we have any constant that stores the zip suffix

Reply:
Not at the moment, but I'm happy to create one if it's important 🙂

Reply:
Resolved with the commit a3528e0 please review